### PR TITLE
Replace ha-clickable-list-item with native material 3 components

### LINKED ISSUE #20865: Integration startup time dialog still uses rounded icons (unlike elsewhere)
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

A while back the rounding on icons was removed from update entity section (see https://github.com/home-assistant/frontend/pull/17972) and integrations page. 

The "integration startup time" dialog found in the repairs section wasn't changed to match though, still has rounded icons where the updated Home Assistant logo looks weird on it.

Screen Shot 2024-05-24 at 1 26 15 PM

### Describe the behavior you expected

No rounding

### Steps to reproduce the issue

1. Settings > Repairs > Overflow menu > Integration startup time
2. Home Assistant icon is used for various integrations there (Core, Supervisor, Version, Trace, etc) and appears cut off due to rounding

### What version of Home Assistant Core has the issue?

2024.5.4

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
Replace ha-clickable-list-item, either with a ha-list-item-new if embedded within a list or a ha-menu-item if embedded within a menu. I decided to remove the component since it's a normal thing in the new material components. It's a drop in replacement although the link will no longer trigger if you activate with a spacebar. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #20865
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **Refactor**
  - Updated various configuration panels to use new list and menu components for improved consistency and maintainability.
  - Replaced `mwc-list` and `mwc-list-item` with `ha-list-new` and `ha-list-item-new`.
  - Swapped `ha-clickable-list-item` with `ha-menu-item` and `ha-button-menu` with `ha-button-menu-new`.

These changes enhance the user interface by standardizing components across different configuration panels.

### CODE REVIEW

**Review by Quentame — APPROVED** [review]
Seems nice 👍

File: src/panels/config/repairs/integrations-startup-time.ts [hunk 1]
```diff
@@ -20,6 +18,8 @@ import {
 import type { HomeAssistant } from "../../../types";
 import { brandsUrl } from "../../../util/brands-url";
 import { documentationUrl } from "../../../util/documentation-url";
+import "../../../components/ha-list-item";
```
Comment (steverep) [inline src/panels/config/repairs/integrations-startup-time.ts:21]:
```suggestion
import "../../../components/ha-list-new";
```

**Review by steverep — COMMENTED** [review]
Made a small import correction, but still reviewing for a11y.  This is my first detailed look at the M3 lists and it seems they've solved some a11y issues but introduced others. 😞

**Comment by silamon** [comment]
@coderabbitai pause

**Comment by silamon** [comment]
Could you clarify what changes you noticed? This PR isn't really focused on accessibility, but if makes things worse, I'm not really going forward with it.

**Comment by steverep** [comment]
> Could you clarify what changes you noticed?

Menu items no longer activating on space key is a significant regression, especially since it sort of appears to activate by closing the menu.  I don't know how they could miss such basic a11y there.

The static lists have ARIA that is just plain wrong (native anchors with a non-interactive `role="listitem"` and no container list role).  The regression on those is the fact that the link role is no longer exposed to assistive technology.

I've briefly tried messing with the attributes to fix but not having success.  I need to look at their code and probably file some issues if they don't exist already.

**Comment by silamon** [comment]
The spacebar thing is something that has never been a part of the material spec, I think we added it in as custom logic: 
https://github.com/home-assistant/frontend/blob/dev/src/components/ha-clickable-list-item.ts#L29

I will set the "interactive" on the lists where possible and see if I can make it have the right ARIA roles. It's also something we can set as an attribute as far as I'm aware.

**Comment by steverep** [comment]
> The spacebar thing is something that has never been a part of the material spec, I think we added it in as custom logic: 

Material is irrelevant; it is part of W3C guidelines and best practices.

> I will set the "interactive" on the lists where possible and see if I can make it have the right ARIA roles. It's also something we can set as an attribute as far as I'm aware.

The `interactive` attribute was removed in favor of the `type` attribute, and I've already tried that.

**Comment by silamon** [comment]
I've had a look at bringing this back to undrafting but the accessilbility issues are not easy to fix since I'm not sure where I can read about those guidelines. I found that spacebar should expand or unexpand the list of a menu button, which it does. At this time we're sure that accessibility is unacceptable with the current m3 implementation? How is the current implementation accessible of the ha-clickable-list-item component or are there problems with that implementation as well?

**Pull request closed without merging**